go-chaos/internal/saas.go

```diff
@@ -59,11 +61,13 @@ func (c K8Client) isSaaSEnvironment() bool {
 	namespace := c.GetCurrentNamespace()
 	clusterId := strings.TrimSuffix(namespace, "-zeebe")
 	zeebeCrd := schema.GroupVersionResource{Group: "cloud.camunda.io", Version: "v1alpha1", Resource: "zeebeclusters"}
-	resource, err := c.DynamicClient.Resource(zeebeCrd).Get(context.TODO(), clusterId, meta.GetOptions{})
+	resource, err := backoff.RetryWithData(func() (*unstructured.Unstructured, error) {
+		return c.DynamicClient.Resource(zeebeCrd).Get(context.TODO(), clusterId, meta.GetOptions{})
+	}, backoff.NewExponentialBackOff())
 
 	if err != nil {
-		LogInfo("Failed to retrieve SaaS CRD, fallback to self-managed mode. %v", err)
-		return false
+		LogError("Failed to check for SaaS CRD, can't proceed without knowing if this is SaaS or Self-Managed: %v", err)
+		panic(err)
```

Comment:
Let's not panic here :) If you want we can take a look at it together tomorrow.

Reply:
But then what? We don't know if it's SaaS or not

Reply:
My point I wanted to make is the following:

I don't want to have the panics in the internal package, you can see it as the lib or backend. It would be same as a library in java just does exit(1), it is unexpected and not nice. 

I would like to see to return an error and handle this by the consumer of the method. 

I think we have here three cases to distinguish: a) CRD exists - SaaS b) CRD not exist - SM c) error -> return error. If currently not existing also causes an error (we have to find out/check this I guess, then we need to somehow need to distinguish these error cases.

Reply:
Okay that part makes sense to me but I'm not sure how to handle the errors. We check for SaaS when building the k8sclient so very early on for every request. What can we do if that fails?